syntax = "proto3";

package agency;

enum SourceType {
    PDF = 0;
    Image = 1;
    URL = 2;
}

message SearchRequest {
    string query = 1;
}

message Source {
    string url = 1;
    string title = 2;
    string description = 3;
    SourceType source_type = 4;
    map<string, string> metadata = 5;
}

message SearchResponse {
    int32 status = 1;
    string result = 2;
    repeated Source sources = 3;
}

service AgencyService {
    rpc pubmed_bioxriv_web_search(SearchRequest) returns (SearchResponse);
}